src/structures/Message.ts

```diff
@@ -1,30 +1,34 @@
 import type { Client } from '#client/Client';
-import { Base } from '#structures/Base';
-import type { Channel } from '#structures/Channel';
-import type { Guild } from '#structures/Guild';
 import { User } from '#structures/User';
 import { log } from '#utils/logger';
-import type { APIMessage } from 'discord-api-types/v10';
+import type { APIMessage, RESTPostAPIChannelMessageJSONBody, Snowflake } from 'discord-api-types/v10';
 import { fetch } from 'undici';
-
+import { Base } from './Base';
+import type { Channel } from './Channel';
+import type { Guild } from './Guild';
 export class Message extends Base {
-	public constructor(
-		public id: string,
-		public content: string,
-		public guild: Guild,
-		public channel: Channel,
-		public author: User,
-		client: Client,
-		public reference?: string
-	) {
+	public id: Snowflake = this.data.id;
+	public content: string = this.data.content;
+	public guild?: Guild = this.data.guild_id ? this.client.guilds.get(this.data.guild_id) : undefined;
+	public channel?: Channel = this.client.channels.get(this.data.channel_id);
+	public author?: User = new User(
```

Comment:
```suggestion
	public author: User = new User(
```
Author shouldnt be nullable, a message always has an author

Reply:
The User object can also not be cached, this would have to be changed in a new PR that revamps the way we store the User object.

Reply:
The data recieved from dapi for a message will provide a user object, so you could cache then if not already, it should still be reworked in the case that the message is a webhook message

Reply:
The preview it gave me didn't give me enough scope sorry,

![image](https://user-images.githubusercontent.com/69066026/164146442-99d8ccb2-0f69-4908-9e6d-8b55aadc4fb1.png)

fixed in https://github.com/TeamAilurus/ailurus/pull/18/commits/2feae306589b8f1673b51782a293f1fc2f45027a